ops/testing.py

```diff
@@ -1191,6 +1190,196 @@ def _get_backend_calls(self, reset: bool = True) -> List[Tuple[Any, ...]]:
             self._backend._calls.clear()
         return calls
 
+    def add_model_secret(self, app_or_unit: AppUnitOrName, content: Dict[str, str]) -> str:
+        """Add a secret owned by the remote application or unit specified.
+
+        This is named :code:`add_model_secret` instead of :code:`add_secret`
+        to distinguish it from the :meth:`ops.model.Application.add_secret`
+        and :meth:`ops.model.Unit.add_secret` methods used by secret owner
+        charms.
+
+        Args:
+            app_or_unit: The remote application (or specific remote unit) that
+                will own the secret.
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+
+        Return:
+            The ID of the newly-secret added.
+        """
+        owner_name = _get_app_or_unit_name(app_or_unit)
+        model.Secret._validate_content(content)
+        return self._backend._secret_add(content, owner_name)
```

Comment:
I feel like a secret object could be returned to the user instead of having them manipulate the secret ID. 

```python
def test_1():
    secret = self.harness.add_secret(owner="remote-app-name", content={"a", "b"})
    secret.grant(receiver="app-under-test-name")
    secret.revoke()
```

This being said, I understand this situation is not specific to secrets and that relations follow the same approach. But I hold the same opinion for relations (it would also be cleaner to access relation data using `relation.data` instead of `get_relation_data(id=123)`)

Reply:
Yeah, I played with this while writing it, but I decided against it for two reasons:

1) As you've noted, other similar harness calls like `get_relation_data` don't do this.
2) It's confusing to have a `testing.Secret` object that's somewhat similar to the `model.Secret` class, but also different enough to be confusing (and the methods have slightly different parameters/types).

So overall I think this is a reasonable approach that fits the current methods.